src/data/dataset.py

```diff
@@ -101,3 +103,59 @@ def collate_fn(self, samples: list[Dialog], return_all_candidates: bool = True):
         b_candidates = b_candidates.view(len(samples), -1, b_candidates.size(1))
 
         return b_contexts, b_candidates.squeeze(1)
+
+
+class CommonsenseConversationDataset(Dataset):
+    def __init__(self, path, tokenizer_name, max_context_len, max_target_len=None, have_candidates=True):
+        self.dataset = []
+
+        self.context_tokenizer = AutoTokenizer.from_pretrained(tokenizer_name, truncation_side="left")
+        self.candidate_tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
+
+        self.max_context_len = max_context_len
+        self.max_target_len = max_target_len or max_context_len
+
+        self.have_candidates = have_candidates
+        self.vocab_size = self.context_tokenizer.vocab_size
+
+        logger.info(f"Loading dataset from '{path}'")
+
+        with open(path, "r") as f:
+            lines_list = list(f)
+
+        for line in lines_list:
+            result = json.loads(line)
+            dialog = {"context": [result["src"]], "candidates": [result["trg"]]}
+            self.dataset.append(Dialog(**dialog))
```

Comment:
Do you have enough memory to load the entire data set? We usually use an iterative dataset that loads the dataset row by row on request.